km/km_init_guest.c

```diff
@@ -329,14 +329,23 @@ int km_clone(km_vcpu_t* vcpu,
    return km_vcpu_get_tid(new_vcpu);
 }
 
-long km_clone3(km_vcpu_t* vcpu, struct clone_args* cl_args)
+long km_clone3(km_vcpu_t* vcpu, struct clone_args* cl_args, size_t cl_args_len)
 {
-   if (cl_args->set_tid_size != 0 || cl_args->cgroup != 0) {
+#ifdef CLONE_ARGS_SIZE_VER2
+   if ((cl_args_len >= CLONE_ARGS_SIZE_VER1 && cl_args->set_tid_size != 0) ||
+       (cl_args_len >= CLONE_ARGS_SIZE_VER2 && cl_args->cgroup != 0)) {
```

Comment:
cgroup is an fd, fd 0 is valid.
Seems like you should check the CLONE_INTO_CGROUP bit in the flags field to decide if they are using cgroup.

Reply:
changed